### PR TITLE
extract duplicated code into super class

### PR DESCRIPTION
**Pull request by tkasparek**
Removing duplicated code from 4 places and putting it into one place.

### CODE REVIEW

File: webapp/app.py [hunk 1]
```diff
@@ -143,7 +181,7 @@ def get(self, nevra=None): # pylint: disable=arguments-differ
         yield self.flush()
 
 
-class UpdatesHandlerPost(BaseHandler):
+class UpdatesHandlerPost(PostHandler):
     """Handler for processing /updates POST requests."""
 
     @gen.coroutine
```
Comment (grinrag) [inline webapp/app.py]: Take a look at http://www.tornadoweb.org/en/stable/gen.html#tornado.gen.coroutine
`From the caller’s perspective, @gen.coroutine is similar to the combination of @return_future and @gen.engine.`
But this method doesn't return future, do it?

**Comment by grinrag** [comment]
Pls checkout to this and look at the code 8e942872adfba3e2657f8d3b308d79533cb62a14

File: webapp/app.py (hunk 1 above)
Comment (grinrag) [reply webapp/app.py]: If you are going to remove this decorator, make sure it works as expected. You can run simple performance test:
`siege -c 100 -r 10 -H "accept: application/json" 'http://127.0.0.1:8080/api/v1/updates POST < pack.json'`

where `pack.json` is a json with list of nevras

**Comment by tkasparek** [comment]
I've moved handle_post() into BaseHandler, thus removed the PostHandler class.

I've also removed the @gen.coroutine from the places where it wasn't needed.

Regarding performance I did not notice any perfromance drop, all of the upstream and refactored code (both with and without @gen.coroutine) are giving me same number of transactions.

**Pull request merged**